recipes/tk/all/conanfile.py

```diff
@@ -0,0 +1,244 @@
+
+            if build_system != "win":
+                # When disabling 64-bit support (in 32-bit), this test must be 0 in order to use "long long" for 64-bit ints
+                # (${tcl_type_64bit} can be either "__int64" or "long long")
+                tools.replace_in_file(os.path.join(config_dir, "configure"),
+                                      "(sizeof(${tcl_type_64bit})==sizeof(long))",
+                                      "(sizeof(${tcl_type_64bit})!=sizeof(long))")
+
+            makefile_in = os.path.join(config_dir, "Makefile.in")
+            # Avoid clearing CFLAGS and LDFLAGS in the makefile
+            # tools.replace_in_file(makefile_in, "\nCFLAGS{}".format(" " if (build_system == "win" and name == "tcl") else "\t"), "\n#CFLAGS\t")
+            tools.replace_in_file(makefile_in, "\nLDFLAGS\t", "\n#LDFLAGS\t")
+            tools.replace_in_file(makefile_in, "${CFLAGS}", "${CFLAGS} ${CPPFLAGS}")
+
+        rules_ext_vc = os.path.join(self.source_folder, self._source_subfolder, "win", "rules-ext.vc")
+        tools.replace_in_file(rules_ext_vc,
+                              "\n_RULESDIR = ",
+                              "\n_RULESDIR = .\n#_RULESDIR = ")
+        rules_vc = os.path.join(self.source_folder, self._source_subfolder, "win", "rules.vc")
+        tools.replace_in_file(rules_vc,
+                              r"$(_TCLDIR)\generic",
+                              r"$(_TCLDIR)\include")
+        tools.replace_in_file(rules_vc,
+                              "\nTCLSTUBLIB",
+                              "\n#TCLSTUBLIB")
+        tools.replace_in_file(rules_vc,
+                              "\nTCLIMPLIB",
+                              "\n#TCLIMPLIB")
+
+        win_makefile_in = os.path.join(self._get_configure_folder("win"), "Makefile.in")
+        tools.replace_in_file(win_makefile_in, "\nTCL_GENERIC_DIR", "\n#TCL_GENERIC_DIR")
+
+        tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "win", "rules.vc"),
+                              "\ncwarn = $(cwarn) -WX",
+                              "\n# cwarn = $(cwarn) -WX")
+
+    def _get_default_build_system(self):
+        if tools.is_apple_os(self.settings.os):
+            return "macosx"
+        elif self.settings.os in ("Linux", "FreeBSD"):
+            return "unix"
+        elif self.settings.os == "Windows":
+            return "win"
+        else:
+            raise ValueError("tk recipe does not recognize os")
+
+    def _get_configure_folder(self, build_system=None):
+        if build_system is None:
+            build_system = self._get_default_build_system()
+        if build_system not in ["win", "unix", "macosx"]:
+            raise ConanExceptionInUserConanfileMethod("Invalid build system: {}".format(build_system))
+        return os.path.join(self.source_folder, self._source_subfolder, build_system)
+
+    def _build_nmake(self, target="release"):
+        # https://core.tcl.tk/tips/doc/trunk/tip/477.md
```

Comment:
Should there be a check for windows and/or msvc in this function?

Reply:
The compiler is checked before calling this method (in `build` and `package`).

I think, in theory, nmake can build with gcc.
Nobody sane would do that.
But it's possible.